### PR TITLE
Updated ktlint version. Fixes #2488

### LINKED ISSUE #2488: Update to latest version of ktlint
We are using ktlint 0.14.0. The latest version is 0.21.0.

Before we can switch we need to fix the issues that the new version has found in our code base.

### PR DESCRIPTION
**Pull request by daivat15**
ktlint version has been updated to fix [#2488](https://github.com/mozilla-mobile/focus-android/issues/2488).
The errors thrown have also been fixed.

### CODE REVIEW

**Comment by pocmo** [comment]
Thank you @daivat15!

Can you update your commit messages to follow this pattern:
> Issue #???: Description of change

File: app/src/main/java/org/mozilla/focus/autocomplete/AutocompletePreference.kt [hunk 1]
```diff
@@ -25,8 +25,7 @@ class AutocompletePreference(context: Context?, attrs: AttributeSet?) : Preferen
 
         summaryView = view?.findViewById(android.R.id.summary)
         summaryView?.setText(
-                if (settings.shouldAutocompleteFromShippedDomainList()
-                        || settings.shouldAutocompleteFromCustomDomainList())
+                if (settings.shouldAutocompleteFromShippedDomainList() || settings.shouldAutocompleteFromCustomDomainList())
```
Comment (sputnikci) [inline app/src/main/java/org/mozilla/focus/autocomplete/AutocompletePreference.kt]: \[detekt\] INFO: \[style/MaxLineLength\] Line detected that is longer than the defined maximum line length in the code style\.

**Comment by pocmo** [comment]
@daivat15 Awesome, thank you! There's one detekt violation that needs to be resolved. And then this might be ready to go.

**Comment by daivat15** [comment]
@pocmo I have fixed it. Please review it when possible.

**Pull request merged**